frontend/src/api/user.ts

```diff
@@ -0,0 +1,86 @@
+import { AxiosResponse } from "axios";
+import { SolvedQuestion } from "../models/SolvedQuestion.model"; // Import the SolvedQuestion model
+import { userServiceClient } from "./server";
+
+/**
+ * Fetch questions completed by the user.
+ * @param userId - The ID of the user for whom you want to fetch completed questions.
+ * @returns An array of completed questions.
+ */
+export async function fetchUserCompletedQuestions(
+  userId: String
+): Promise<SolvedQuestion[]> { 
+  try {
+    const response: AxiosResponse = await userServiceClient.get(
+      `/api/users/${userId}/questions`
+    );
+
+    const resData = response.data;
+    const completedQuestions: SolvedQuestion[] = resData.map(
+      (q: any) =>
+        new SolvedQuestion(
+          q._id,
+          q.id,
+          q.title,
+          q.description,
+          q.topics,
+          q.difficulty,
+          false, // Default value for solved
+          undefined // Default value for solvedDate
+        )
+    );
+
+    return completedQuestions;
+  } catch (error) {
+    // Handle errors appropriately, e.g., log the error, return an empty array, or throw an error.
+    console.error(error);
+    throw error;
+  }
+}
+
+/**
+ * Add a new question to the user's completed questions.
+ * @param userId - The ID of the user to whom you want to add the question.
+ * @param questionId - The ID of the question you want to add.
+ * @param complexity - The complexity of the question.
+ * @param category - The category of the question.
+ * @returns The added question.
+ */
+export async function addUserQuestion(
+  userId: number,
+  questionId: string,
+  complexity: number,
+  category: string[]
+): Promise<SolvedQuestion> { 
+  try {
+    const response: AxiosResponse = await userServiceClient.post(
+      `/api/users/${userId}/addquestions`,
+      {
+        userId,
+        questionId,
+        complexity,
+        category,
+      }
```

Comment:
https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g10/pull/43/files#r1375639810

As mentioned in this comment, please make sure you supply the necessary data (the `solved` field?) to "add" a question to user's question history